### PR TITLE
single precision support in skimage.segmentation

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR adds float32 support to most functions in `skimage.segmentation`.

`random_walker` and `slic` already had float32 support, but some additional tests were added.

I skipped `felzenswalb` and `watershed` for now as the Cython code for these was a little more involved, so those two still promote to double precision internally.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/segmentation/active_contour_model.py [hunk 1]
```diff
@@ -134,21 +140,23 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
-        6*np.eye(n)  # fourth order derivative, central difference
-    A = -alpha*a + beta*b
+    xsave = np.empty((convergence_order, n), dtype=float_dtype)
+    ysave = np.empty((convergence_order, n), dtype=float_dtype)
+
+    # Build snake shape matrix for Euler equation in double precision
+    eye_n = np.eye(n, dtype=float)
+    a = np.roll(eye_n, -1, axis=0) + \
+        np.roll(eye_n, -1, axis=1) - \
+        2 * eye_n  # second order derivative, central difference
```
Comment (rfezzani) [inline skimage/segmentation/active_contour_model.py:153]:
```suggestion
    a = (np.roll(eye_n, -1, axis=0)
         + np.roll(eye_n, -1, axis=1)
         - 2 * eye_n)  # second order derivative, central difference
```

File: skimage/segmentation/active_contour_model.py [hunk 2]
```diff
@@ -134,21 +140,23 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
+    b = np.roll(eye_n, -2, axis=0) + \
+        np.roll(eye_n, -2, axis=1) - \
+        4 * np.roll(eye_n, -1, axis=0) - \
+        4 * np.roll(eye_n, -1, axis=1) + \
+        6 * eye_n  # fourth order derivative, central difference
```
Comment (rfezzani) [inline skimage/segmentation/active_contour_model.py:158]:
```suggestion
    b = (np.roll(eye_n, -2, axis=0)
         + np.roll(eye_n, -2, axis=1)
         - 4 * np.roll(eye_n, -1, axis=0)
         - 4 * np.roll(eye_n, -1, axis=1)
         + 6 * eye_n)  # fourth order derivative, central difference
```

**Review by rfezzani — APPROVED** [review]
Excellent @grlee77, I just left a code formatting suggestion :wink:. Thank you!

**Review by mkcor — APPROVED** [review]
Thank you for all the cleanup as well!

**Pull request merged**